server/http.go

```diff
@@ -38,116 +31,75 @@ const (
 }
 
-func handleHTTPRequest(p *Plugin, w http.ResponseWriter, r *http.Request) (int, error) {
-	switch r.URL.Path {
-	// Issue APIs
-	case routeAPICreateIssue:
-		return withInstance(p, w, r, httpAPICreateIssue)
-	case routeAPIGetCreateIssueMetadata:
-		return withInstance(p, w, r, httpAPIGetCreateIssueMetadata)
-	case routeAPIAttachCommentToIssue:
-		return withInstance(p, w, r, httpAPIAttachCommentToIssue)
-
-	// User APIs
-	case routeAPIUserInfo:
-		return withInstance(p, w, r, httpAPIGetUserInfo)
-
-	// Atlassian Connect application
-	case routeACInstalled:
-		return httpACInstalled(p, w, r)
-	case routeACJSON:
-		return httpACJSON(p, w, r)
-	case routeACUninstalled:
-		return httpACUninstalled(p, w, r)
-
-	// Atlassian Connect user mapping
-	case routeACUserRedirectWithToken:
-		return withCloudInstance(p, w, r, httpACUserRedirect)
-	case routeACUserConfirm,
-		routeACUserConnected,
-		routeACUserDisconnected:
-		return withCloudInstance(p, w, r, httpACUserInteractive)
-
-	// Incoming webhook
-	case routeIncomingWebhook, routeIncomingIssueEvent:
-		return httpWebhook(p, w, r)
-
-	// Oauth1 (Jira Server)
-	case routeOAuth1Complete:
-		return withServerInstance(p, w, r, httpOAuth1Complete)
-	case routeOAuth1PublicKey:
-		return httpOAuth1PublicKey(p, w, r)
-
-	// User connect/disconnect links
-	case routeUserConnect:
-		return withInstance(p, w, r, httpUserConnect)
-	case routeUserDisconnect:
-		return withInstance(p, w, r, httpUserDisconnect)
-
-	// Firehose webhook setup for channel subscriptions
-	case routeAPISubscribeWebhook:
-		return httpSubscribeWebhook(p, w, r)
-	}
-
-	if strings.HasPrefix(r.URL.Path, routeAPISubscriptionsChannel) {
-		return httpChannelSubscriptions(p, w, r)
-	}
-
-	return http.StatusNotFound, errors.New("not found")
+func httpGetFilter(ff ...ActionFunc) ActionFilter {
+	return append(ActionFilter{RequireHTTPGet}, ff...)
 }
 
-func (p *Plugin) loadTemplates(dir string) (map[string]*template.Template, error) {
-	templates := make(map[string]*template.Template)
-	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
-		if err != nil {
-			return err
-		}
-		if info.IsDir() {
-			return nil
-		}
-		template, err := template.ParseFiles(path)
-		if err != nil {
-			p.errorf("OnActivate: failed to parse template %s: %v", path, err)
+var httpInstanceFilter = ActionFilter{RequireHTTPMattermostUserId, RequireInstance}
+var httpJiraUserFilter = ActionFilter{RequireHTTPMattermostUserId, RequireInstance, RequireJiraUser}
+var httpJiraClientFilter = append(httpJiraUserFilter, RequireJiraClient)
+
+var httpRouter = ActionRouter{
+	DefaultRouteHandler: func(a *Action) error {
+		return a.RespondError(http.StatusNotFound, nil, "not found")
+	},
+	Log: ActionFilter{
+		func(a *Action) error {
+			if a.HTTPStatusCode == 0 {
+				a.HTTPStatusCode = http.StatusOK
+			}
+			if a.Err != nil {
+				a.Plugin.errorf("http: %v %s %v", a.HTTPStatusCode, a.HTTPRequest.URL.String(), a.Err)
```

Comment:
This should not be logging at error level by default. See https://community.mattermost.com/core/pl/mx5rdy5wyifh7d9s6nxu3gpkpa

Reply:
done